drivers/regulator/regulator_da1469x.c

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright 2023 Renesas Electronics Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_smartbond_regulator
+
+#include <stdint.h>
+
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/logging/log.h>
+#include <DA1469xAB.h>
+
+LOG_MODULE_REGISTER(regulator_da1469x, CONFIG_REGULATOR_LOG_LEVEL);
+
+/* Rail should be powered during sleep, mode applies to VDD, V18, V18P */
+#define DA1469X_MODE_SLEEP_LDO					0x01
+/* Settings valid for VDD, V14, V18, V18P */
+#define DA1469X_MODE_DCDC_HIGH_BATTERY_VOLTAGE_ON		0x02
+#define DA1469X_MODE_DCDC_LOW_BATTERY_VOLTAGE_ON		0x04
+/* Settings valid for V30 */
+#define DA1469X_MODE_LDO_VBUS_ON				0x08
+#define DA1469X_MODE_LDO_VBAT_ON				0x10
+#define DA1469X_MODE_V30_VBAT_CLAMP_ON				0x20
+
+#define DCDC_REG_FIELD_MSK(dcdc, field) \
```

Comment:
Have these macros come from an SDK, or were they written specifically for this driver?
I find them rather difficult to follow.

I think this would be better approached with a description structure for each regulator containing appropriate masks and fields, e.g:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/regulator/regulator_pca9420.c

Reply:
All macros removed, pure SDK fields used across file